libs/stats/odc/stats/tasks.py

```diff
@@ -256,9 +285,13 @@ def __init__(self,
         self._gridspec = gridspec
         self._all_tiles = sorted(idx for idx, _ in cache.tiles(grid))
 
+    def _delete_local_cache(self):
```

Comment:
Is this supposed to be called from destructor? Does it need a function of it's own?

```python
    def __del__(self):
        # co cleanup
```

Reply:
Also, is this cleaning up the dscache file after every tile? It really should cache the cache forever.

Reply:
@alexgleith are you suggesting we leave the file and it gets destroyed when the pod gets destroyed?

Reply:
@alexgleith download is per app invocation, not per tile.

Reply:
Ok good, so long as we're not just caching it for one job!